src/main/java/uk/gov/hmcts/reform/userprofileapi/service/UserProfileUpdator.java

```diff
@@ -28,42 +41,91 @@
     @Autowired
     private AuditRepository auditRepository;
 
+    @Autowired
+    private IdamFeignClient idamClient;
+
     @Override
     public UserProfile update(UpdateUserProfileData updateUserProfileData, String userId) {
 
         HttpStatus status = HttpStatus.OK;
         UserProfile userProfile = null;
         if (!isUserIdValid(userId, false)) {
-            persistAudit(HttpStatus.NOT_FOUND, null);
+            persistAudit(HttpStatus.NOT_FOUND, null, ResponseSource.SYNC);
             throw new ResourceNotFoundException("userId provided is malformed");
         }
 
         Optional<UserProfile> userProfileOptional = userProfileRepository.findByIdamId(userId);
         userProfile =  userProfileOptional.orElse(null);
 
         if (userProfile == null) {
-            persistAudit(HttpStatus.NOT_FOUND, null);
+            persistAudit(HttpStatus.NOT_FOUND, null, ResponseSource.SYNC);
             throw new ResourceNotFoundException("could not find user profile for userId: " + userId);
         } else if (!isUpdateUserProfileRequestValid(updateUserProfileData)) {
-            persistAudit(HttpStatus.BAD_REQUEST, null);
+            persistAudit(HttpStatus.BAD_REQUEST, null,ResponseSource.SYNC);
             throw new RequiredFieldMissingException("Update user profile request is not valid for userId: " + userId);
         } else if (!isSameAsExistingUserProfile(updateUserProfileData, userProfile)) {
+
             userProfile.setEmail(updateUserProfileData.getEmail().trim());
             userProfile.setFirstName(updateUserProfileData.getFirstName().trim());
             userProfile.setLastName(updateUserProfileData.getLastName().trim());
             userProfile.setStatus(IdamStatus.valueOf(updateUserProfileData.getIdamStatus().toUpperCase()));
-            try {
-                userProfile = userProfileRepository.save(userProfile);
-            } catch (Exception ex) {
-                status = HttpStatus.INTERNAL_SERVER_ERROR;
-            }
         }
-        persistAudit(status, userProfile);
+
+        try {
+            userProfile = userProfileRepository.save(userProfile);
+        } catch (Exception ex) {
+            status = HttpStatus.INTERNAL_SERVER_ERROR;
+        }
+        persistAudit(status, userProfile, ResponseSource.SYNC);
         return userProfile;
     }
 
-    public void persistAudit(HttpStatus idamStatus, UserProfile userProfile) {
-        Audit audit = new Audit(idamStatus.value(), resolveStatusAndReturnMessage(idamStatus), ResponseSource.SYNC, userProfile);
+    @Override
+    public UserProfileRolesResponse updateRoles(UpdateUserProfileData profileData, String userId) {
```

Comment:
This is very hard-to-read and confusing code. Consider refactoring.

Reply:
Removed multi return statement